mitosheet/src/components/taskpanes/UpdateImports/UpdateImportsTaskpane.tsx

```diff
@@ -0,0 +1,236 @@
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneFooter from "../DefaultTaskpane/DefaultTaskpaneFooter";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+import { CSVImportParams } from "../../import/CSVImportScreen";
+import { ExcelImportParams } from "../../import/XLSXImportScreen";
+import ImportCard from "./UpdateImportCard";
+import { useStateFromAPIAsync } from "../../../hooks/useStateFromAPIAsync";
+import { getBaseOfPath, isCSVImportData, isCSVImportParams, isExcelImportData, isExcelImportParams } from "./UpdateImportsUtils";
+import { ImportScreen } from "../FileImport/FileImportTaskpane";
+import UpdateDataframeImportTaskpane from "./UpdateDataframeImportTaskpane";
+import UpdateXLSXImportsTaskpane from "./UpdateXLSXImportTaskpane";
+import UpdateCSVImportTaskpane from "./UpdateCSVImportTaskpane";
+import UpdateFileBrowserTaskpane from "./UpdateFileBrowserTaskpane";
+
+
+interface updateImportsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    userProfile: UserProfile;
+    analysisData: AnalysisData;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+
+    currPathParts: string[];
+    setCurrPathParts: (newCurrPathParts: string[]) => void;
+
+    sheetDataArray: SheetData[];
+    selectedSheetIndex: number;
+}
+
+export interface CSVImportData {
+    step_type: 'simple_import',
+    params: CSVImportParams
+}
+export interface ExcelImportData {
+    step_type: 'excel_import',
+    params: ExcelImportParams
+}
+export interface DataframeImportData {
+    step_type: 'dataframe_import',
+    params: DataframeImportParams
+}
+
+export interface StepImportData {
+    step_id: string,
+    imports: (CSVImportData | ExcelImportData | DataframeImportData)[];
+}
+
+export interface DataframeCreationIndex {step_id: string, index: number}
+export interface ReplacingDataframeState {
+    dataframeCreationIndex: DataframeCreationIndex, 
+    screen: ImportScreen | 'dataframe_import',
+    params: CSVImportParams | ExcelImportParams | DataframeImportParams | undefined
+}
+
+    
+
+/* 
+    This is the updateImports taskpane.
+*/
+const UpdateImportsTaskpane = (props: updateImportsTaskpaneProps): JSX.Element => {
+
+    const [updatedStepImportData, setUpdatedStepImportData] = useState<StepImportData[]>([]);
+    const [displayedImportCardDropdown, setDisplayedImportCardDropdown] = useState<DataframeCreationIndex | undefined>(undefined);
+    const [replacingDataframeState, setReplacingDataframeState] = useState<ReplacingDataframeState | undefined>(undefined);
+
+    const [originalStepImportData] = useStateFromAPIAsync(
+        undefined,
+        () => {return props.mitoAPI.getImportedFilesAndDataframes()},
+        (loadedData) => {
+            // On load, update the updated import data
+            setUpdatedStepImportData(loadedData || [])
+        }
+    )
+
```

Comment:
I don't see a distinction between the `updatedStepImportData` and `originalStepImportData`. Currently, they are the same thing as we are setting the updatedStepImportData during the `useStateFromAPIAysnc` call. 

After I update a file import, the `originalStepImportData` is also getting updated somewhere